From d1193b25155fd7410c855abdf7c126690aa91dcf Mon Sep 17 00:00:00 2001
From: Yuanzheng Song <songyuanzheng@huawei.com>
Date: Tue, 17 May 2022 02:05:20 +0000
Subject: [PATCH 2672/4585] [Huawei] gcc-plugins: change the $HOSTCC to the $CC

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 186766

--------------------------------

If $HOSTCC is used to determine whether to enable GCC_PLUGINS,
compilation errors occur on some devices.
So fix it by changing the $HOSTCC to the $CC.

Fixes: 46d62946fc12 ("[Huawei] Revert "[Backport] gcc-plugins: simplify GCC plugin-dev capability test"")
Signed-off-by: Yuanzheng Song <songyuanzheng@huawei.com>
---
 scripts/gcc-plugin.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/gcc-plugin.sh b/scripts/gcc-plugin.sh
index b79fd0bea838..86d8bead6ab2 100755
--- a/scripts/gcc-plugin.sh
+++ b/scripts/gcc-plugin.sh
@@ -8,7 +8,7 @@ srctree=$(dirname "$0")
 gccplugins_dir=$($* -print-file-name=plugin)
 
 # we need a c++ compiler that supports the designated initializer GNU extension
-$HOSTCC -c -x c++ -std=gnu++98 - -fsyntax-only -I $srctree/gcc-plugins -I $gccplugins_dir/include 2>/dev/null <<EOF
+$CC -c -x c++ -std=gnu++98 - -fsyntax-only -I $srctree/gcc-plugins -I $gccplugins_dir/include 2>/dev/null <<EOF
 #include "gcc-common.h"
 class test {
 public:
-- 
2.17.1

